text-motions.c

```diff
@@ -586,25 +586,37 @@ size_t text_function_end_prev(Text *txt, size_t pos) {
 	return text_function_end_direction(txt, pos, -1);
 }
 
-size_t text_block_start(Text *txt, size_t pos) {
+static size_t text_block_direction(Text *txt, size_t pos, int direction) {
 	int nest = 0;
-	char c;
+	char c, target, nest_target;
 	bool instring = false;
 	Iterator it = text_iterator_get(txt, pos);
-	while (text_iterator_byte_prev(&it, &c)) {
+
+	target = (direction > 0 ? '}' : '{');
+	nest_target = (direction > 0 ? '{' : '}');
```

Comment:
I don’t have any issues with the parens motion, which is useful in general enough contexts, but a “block” motion only applicable to a subset of contexts is one of those Vim legacy misfeatures that got me interested in alternate implementations in the first place.